mock/py/mockbuild/backend.py

```diff
@@ -583,14 +584,25 @@ def buildsrpm(self, spec, sources, timeout, follow_links):
                 # Resolve any symlinks
                 sources = os.path.realpath(sources)
                 if os.path.isdir(sources):
-                    util.rmtree(self.buildroot.make_chroot_path(self.buildroot.builddir, "SOURCES"))
+                    host_chroot_sources = self.buildroot.make_chroot_path(
+                        self.buildroot.builddir, "SOURCES")
+
+                    util.rmtree(host_chroot_sources)
                     shutil.copytree(sources,
-                                    self.buildroot.make_chroot_path(self.buildroot.builddir, "SOURCES"),
+                                    host_chroot_sources,
                                     symlinks=(not follow_links))
                 else:
-                    shutil.copy(sources, self.buildroot.make_chroot_path(self.buildroot.builddir, "SOURCES"))
+                    host_chroot_sources = self.buildroot.make_chroot_path(
+                        self.buildroot.builddir, "SOURCES", os.path.basename(sources))
+                    shutil.copy(sources, host_chroot_sources)
 
             spec = self.buildroot.make_chroot_path(self.buildroot.builddir, "SPECS", os.path.basename(spec))
+
+            self.plugins.call_hooks('pre_srpm_build',
+                                    spec,
+                                    host_chroot_sources,
+                                    self.private_network)
```

Comment:
I'd drop the `self.private_network` parameter, you can use equivalent config `rpmbuild_networking`.

Reply:
I went for the `self.private_network` parameter because it is used everywhere in operations like `buildroot.doChroot`. I was worried that you may change the definition of `self.private_network` to be something else than `not config['rpmbuild_networking']` in future and the change wouldn't propagate to plugin.

Reply:
I wouldn't be afraid of that;  if that happened, there would have to be
compat fix for the plugin.  And making the plugin api as simple as
possible has preference here (the less parameters in hook the better).

Reply:
I added the new commit not to pass `private_network` into the plugin.